### PR TITLE
Enable WebSocket Affinity in the device agent

### PR DESCRIPTION
**Pull request by hardillb**
part of FlowFuse/flowfuse#2514

## Description

As part of work to enable horizontal scaling of the Forge App, this allows the Device agent to use a cookie to bind to a specific instance (using Nginx Affinity mapping to start with).

Reuses the affinity cookie if present, shares the affinity cookie with the forge platform if available.

## Related Issue(s)

FlowFuse/flowfuse#2514

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

**Comment by hardillb** [comment]
Need to work out how to test this

**Comment by hardillb** [comment]
I've used the default Cookie name here, we may want to set it to a FF specific value.

**Review by knolleary — COMMENTED** [review]
As proposed in https://github.com/FlowFuse/flowfuse/pull/3173/files#r1445027729 - lets use `FFSESSION` for the cookie name

File: lib/editor/tunnel.js [hunk 1]
```diff
@@ -57,9 +58,24 @@ class EditorTunnel {
...
+        }
         const socket = newWsConnection(forgeWSEndpoint, {
-            headers: {
-                'x-access-token': this.options.token
+            headers
+        })
+        socket.on('upgrade', (evt) => {
+            if (evt.headers && evt.headers['set-cookie']) {
+                const cookies = evt.headers['set-cookie']
+                cookies.forEach(cookie => {
```
Comment (knolleary) [inline lib/editor/tunnel.js:73]: @hardillb Are we certain that `evt.headers['set-cookie']` is always going to be an array? I don't readily have the means to verify this locally.

Comment (hardillb) [reply lib/editor/tunnel.js:73]: I'll double check the WS code, but if I can't find a definitive answer I'll add some defensive code

**Review by knolleary — APPROVED** [review]

**Pull request merged**